ddl/partition.go

```diff
@@ -124,18 +141,16 @@ func buildRangePartitionDefinitions(ctx sessionctx.Context, d *ddl, s *ast.Creat
 		if err != nil {
 			return errors.Trace(err)
 		}
+		comment, _ := def.Comment()
 		piDef := model.PartitionDefinition{
 			Name:    def.Name,
 			ID:      pid,
-			Comment: def.Comment,
+			Comment: comment,
 		}
 
-		if s.Partition.ColumnNames == nil && len(def.LessThan) != 1 {
```

Comment:
We need this restriction.
Currently, `partition by range columns` could work when there is only one column.

Reply:
This should already been checked in the parser. https://github.com/pingcap/parser/blob/b55d93a9d8f6a865ff575f4e8558d0f6bae89649/ast/ddl.go#L2069-L2070

You mean we should check again here?